snapcraft.yaml

```diff
@@ -47,26 +47,29 @@ parts:
     plugin: python
     source: certbot
     source-subdir: acme
-    constraints: $SNAPCRAFT_PART_SRC/constraints.txt
+    constraints: [$SNAPCRAFT_PART_SRC/constraints.txt]
     python-version: python3
   certbot:
     plugin: python
     source: certbot
-    constraints: $SNAPCRAFT_PART_SRC/constraints.txt
+    constraints: [$SNAPCRAFT_PART_SRC/constraints.txt]
     python-version: python3
     after: [acme]
+    override-pull: |
```

Comment:
Is there a specific advantage to take the snap version from the certbot part?

Reply:
Yes; I tried to explain this in my commit message from commit af7cccf:

> 
> The use of version-script seems to break with either newer snapcraft,
> snapcraft with lxd or core18 (it's not clear to me which). The breakage
> is related to "parts/certbot/src" not being found. This can be fixed
> with $SNAPCRAFT_PART_SRC, but this doesn't seem to be defined at
> "version-script time".
> 
> However https://snapcraft.io/docs/deprecation-notices/dn10 deprecates
> the use of version-script, and if we convert to the recommended new way,
> then we use override-pull instead and $SNAPCRAFT_PART_SRC is defined
> there, so this conveniently fixes both problems at once.

Reply:
Ok, it is clear for me now !